control-announcements/src/main/java/org/oskari/announcements/actions/AnnouncementsServiceMybatisImpl.java

```diff
@@ -0,0 +1,104 @@
+import fi.nls.oskari.db.DatasourceHelper;
+import org.oskari.announcements.helpers.Announcement;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.service.ServiceRuntimeException;
+import org.json.JSONObject;
+import org.oskari.announcements.helpers.AnnouncementsParser;
+import org.oskari.announcements.mappers.AnnouncementsMapper;
+import java.time.LocalDate;
+
+@Oskari
+public class AnnouncementsServiceMybatisImpl extends AnnouncementsService{
+
+    private static final Logger LOG = LogFactory.getLogger(AnnouncementsServiceMybatisImpl.class);
+
+    private SqlSessionFactory factory = null;
+    private AnnouncementsParser parser = new AnnouncementsParser();
+
+    public AnnouncementsServiceMybatisImpl() {
+        final DatasourceHelper helper = DatasourceHelper.getInstance();
+        DataSource dataSource = helper.getDataSource();
+        if (dataSource == null) {
+            dataSource = helper.createDataSource();
+        }
+        if (dataSource == null) {
+            LOG.error("Couldn't get datasource for oskari announcements service");
+        }
+        factory = initializeMyBatis(dataSource);
+    }
+
+    private SqlSessionFactory initializeMyBatis(final DataSource dataSource) {
+        final TransactionFactory transactionFactory = new JdbcTransactionFactory();
+        final Environment environment = new Environment("development", transactionFactory, dataSource);
+
+        final Configuration configuration = new Configuration(environment);
+        configuration.getTypeAliasRegistry().registerAlias(AnnouncementsService.class);
+        configuration.setLazyLoadingEnabled(true);
+        configuration.addMapper(AnnouncementsMapper.class);
+
+        return new SqlSessionFactoryBuilder().build(configuration);
+    }
+
+    public JSONObject getAdminAnnouncements() {
+        try (final SqlSession session = factory.openSession()) {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            return parser.parseAnnouncementsMap(mapper.getAdminAnnouncements());
+        } catch (Exception e) {
+            throw new ServiceRuntimeException("Failed to get admin announcements", e);
+        }
+    }
+
+    public JSONObject getAnnouncements() {
+        try (final SqlSession session = factory.openSession()) {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            return parser.parseAnnouncementsMap(mapper.getAnnouncements(LocalDate.now()));
+        } catch (Exception e) {
+            throw new ServiceRuntimeException("Failed to get announcements", e);
+        }
+    }
+
+    public int updateAnnouncement(final Announcement announcement) {
+        int updateId = -1;
+        try (final SqlSession session = factory.openSession()) {
+            final AnnouncementsMapper mapper = session.getMapper(AnnouncementsMapper.class);
+            updateId = mapper.updateAnnouncement(announcement);
```

Comment:
You'll probably need that commit() call back on these write ops?

Reply:
Whoopsie, dropped it on accident, let's add those back